File: mobile/ios/Core/Billing/BillingService.swift
import Foundation

struct BillingPackageOption: Sendable, Hashable, Identifiable {
    let id: String
    let planCode: PlanCode
    let interval: BillingInterval
    let localizedPrice: String
}

enum BillingServiceError: LocalizedError, Equatable {
    case notConfigured
    case notAuthenticated
    case offeringUnavailable
    case packageUnavailable

    var errorDescription: String? {
        switch self {
        case .notConfigured: "支付尚未配置"
        case .notAuthenticated: "登录完成后才能购买套餐"
        case .offeringUnavailable: "当前没有可购买的套餐"
        case .packageUnavailable: "所选套餐暂不可购买"
        }
    }
}

enum BillingPurchaseResult: Sendable, Equatable { case purchased, cancelled }

@MainActor
protocol BillingService: AnyObject {
    var isConfigured: Bool { get }
    func identify(userID: UUID) async throws
    func clearIdentity() async
    func fetchPackages() async throws -> [BillingPackageOption]
    func purchase(packageID: String) async throws -> BillingPurchaseResult
    func restorePurchases() async throws
}

File: mobile/android/app/src/main/kotlin/com/codeiy/im/core/billing/BillingService.kt
package com.codeiy.im.core.billing

import android.app.Activity
import com.codeiy.im.model.BillingInterval
import com.codeiy.im.model.PlanCode

data class BillingPackageOption(
    val id: String,
    val planCode: PlanCode,
    val interval: BillingInterval,
    val localizedPrice: String,
)

sealed interface BillingPurchaseResult {
    data object Purchased : BillingPurchaseResult
    data object Cancelled : BillingPurchaseResult
}

class BillingUnavailableException(message: String) : IllegalStateException(message)

interface BillingService {
    val isConfigured: Boolean
    suspend fun identify(userId: String)
    suspend fun clearIdentity()
    suspend fun fetchPackages(): List<BillingPackageOption>
    suspend fun purchase(activity: Activity, packageId: String): BillingPurchaseResult
    suspend fun restorePurchases()
}
